File: lib/features/home/domain/entitites/track.dart
class Track {
  final int trackId;
  final String title;
  final String artist;
  final String image;
  final String source;
  final String album;
  final int favorite;

  Track({
    required this.trackId,
    required this.title,
    required this.artist,
    required this.image,
    required this.album,
    required this.source,
    required this.favorite,
  });
}

File: backend/SoundSpace/Entities/Product/Track.cs
using SoundSpace.Entities.Auth;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SoundSpace.Entities.Product

{
    public class Track
    {

        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int TrackId { get; set; }
        public string Title { get; set; }
        public string Image { get; set; }
        public string Source { get; set; }
        public string Album { get; set; }
        public string Lyric { get; set; }

        // Mối quan hệ 1-1 với User (người đăng bài hát)
        [ForeignKey("User")]
        public int ArtistId { get; set; }
        public User Artist { get; set; }

        public ICollection<TrackPlaylist> Playlists { get; set; } = new List<TrackPlaylist>();

        public ICollection<FavoriteTrack> Favorite { get; set; } = new List<FavoriteTrack>();



    }
}
